### ISSUE TITLE
"Length cannot be less than zero" exception when trimming a record with more than 158 spaces

### ISSUE THREAD

**Issue by Boaz101** [JoshClose/CsvHelper#1715]
**Describe the bug**
Getting an error "Length cannot be less than zero." when reading a csv file with a record that has lots of white space and the Trim option turned on.

**To Reproduce**
If there are 158+ spaces in the csv file record then an exception occurs in the CsvReader.Read() method:
```
            string content = @"Header1,Header2
                                                                                                                                                              ";
            using StringReader stringReader = new StringReader(content);
            var config = new CsvConfiguration(CultureInfo.CurrentCulture)
            {
                TrimOptions = TrimOptions.Trim,
            };
            using var _csvReader = new CsvReader(stringReader, config);

            while (_csvReader.Read())
            { }
```

System.ArgumentOutOfRangeException: 'Length cannot be less than zero. (Parameter 'length')'
   at System.String.Ctor(Char[] value, Int32 startIndex, Int32 length)
   at CsvHelper.CsvParser.get_Record()
   at CsvHelper.CsvReader.Read()

**Expected behavior**
Trim the white space without throwing an exception.

**Additional context**
This issue does not occur in v19. It starts occurring when I upgrade to v20 or greater.

**Comment by JoshClose** [comment]
This is odd. If I run through the parser directly, I don't get the error. When running it through the reader, the parser blows up. I'll figure it out.

**Comment by JoshClose** [comment]
Seems to start at 9 whitespaces actually.

**Comment by JoshClose** [comment]
Fixed in 24.0.1 on NuGet. https://www.nuget.org/packages/CsvHelper/24.0.1

**Comment by Boaz101** [comment]
That was fast! Working well now. Thank you!
